python/across_api/base/ephem.py

```diff
@@ -86,6 +86,11 @@ def ephindex(self, t: Time) -> int:
         -------
             The index of the nearest time in the ephemeris.
         """
+        # Check that the time is within the ephemeris range.
+        assert (
+            t.jd >= self.begin.jd and t.jd <= self.end.jd
+        ), "Time outside of ephemeris of range"
```

Comment:
This assertion is still incorrect. Please assert that the integer index computed in the line below is in bounds.

Reply:
This is how the integer is calculated:

```int(np.argmin(np.abs((self.timestamp.datetime - t.datetime))))```

If `t` is outside of the scope of the Ephemeris, it will just return `0` or the maximum index value. So asserting i < 0 or i > i_max will *never* be triggered.

Reply:
I don't fully understand what `argmin` does here, but in general if you are converting from a floating point value to an index into a bounded array, I would do it like this:

```python
i = int((t - t_min) // dt)
assert i >= 0 && i < i_max
```

If you write it instead like this:

```python
assert t >= t_min && t < t_max
i = int((t - t_min) // dt)
```

then it is likely that the assertion is incorrect.

Reply:
The nuance of using `np.argmin` here so that if you have a time of 1.999999s, and are searching a time series that has 1 second resolution, it finds the closest value, so would return the index for 2s. The way you suggest would return the index for 1s.

Reply:
This... I HOPE... satisfies both of us. [e1ea448](https://github.com/nasa-gcn/gcn.nasa.gov/pull/1975/commits/e1ea448ed4b989433ddc6f02a242c9ee9c639a65)